### ISSUE TITLE
A few feedbacks

### ISSUE THREAD

**Issue by fredmaggiowski** [theliuk/todoapp#1]
* Error handling adds too much overhead, a sentinel error would be much better for this use case;
 * I'd change function names in `TodoService` interface definition removing redundant "Todo":
```
type TodoService interface {
	Get(ID string) (Todo, error)
	Create(t Todo) (string, error)
	Delete(t Todo) error
	Update(t Todo) error
}
```
 * Take care of exported and unexported fields in `inmem/inmem.go` implementation. I'd not export the `Todos` map storage, it'd be better to have a constructor-like function to create an instance of the service and provide the `UniqueIDGenerator` function implementation.

**Comment by theliuk** [comment]
I agree with you for what concerns error handling and the interface's method naming.
I don't really agree with the need of a constructor function to create the service instance.
In the end, the service will be used only by means of its interface; it will the main function to create the service from scratch and pass its reference around, where it will be used only through the interface.

By making the Todo attribute public, I can easily test it without the need of placing the tests inside the same namespace of the service (doing white boxing testing).

**Comment by fredmaggiowski** [comment]
>I don't really agree with the need of a constructor function to create the service instance.
>In the end, the service will be used only by means of its interface; it will the main function to create the service from scratch and pass its reference around, where it will be used only through the interface.

I understand and I'm fine with the choice :)

 > By making the Todo attribute public, I can easily test it without the need of placing the tests inside the same namespace of the service (doing white boxing testing).

This is not correct though, tests should reside with the implementation, not somewhere else or being delegated to other packages. You have defined a public interface, and only that should be exported. 
You're letting developers access something that is not related to the public interface of a `TodoService` (and is also not thread-safe since you're protecting it with a RWMutex :wink:).
